.github/workflows/build-test.yml

```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  workflow_dispatch: { }
+  workflow_call: { }
+
+jobs:
+  build-test-book:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        python-version: [ "3.9" ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Upgrade pip
+        run: |
+          # install pip=>20.1 to use "pip cache dir"
+          python3 -m pip install --upgrade pip setuptools wheel
+
+      - name: Get pip cache dir
+        id: pip-cache
+        run: echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: python3 -m pip install nox
+
+      - name: Build book
+        run: nox -s docs-test
+
+      # Test for bad links and ensure alt tags for usability
+      - name: Check HTML using htmlproofer
+        uses: chabad360/htmlproofer@master
+        with:
+          directory: ./_build/html
```

Comment:
I like this approach. However, if you want to test the links, especially with Sphinx, you can use a link checker [Sphinx linkcheck](https://www.sphinx-doc.org/en/master/usage/builders/index.html#builders). We can integrate it into the Nox command along with the build process, allowing you to identify broken links while building the documentation. This makes debugging easier

Reply:
this sounds good to me. we've had a lot of problems with html proofer lately. the only downside is i think sphinx linkcheck doesn't check for alt tags?